icaruscode/CRT/CRTUtils/CRTHitRecoAlg.cc

```diff
@@ -393,27 +392,41 @@ sbn::crt::CRTHit CRTHitRecoAlg::MakeTopHit(art::Ptr<CRTData> data, ULong64_t Glo
     auto const& adsGeo = adGeo.SensitiveVolume(adsid_max); //trigger strip
     uint64_t thit = data->fTs0;
     Long64_t thit1 = data->fTs1;
-
-    if(adsid_max<8){
-        thit -= (uint64_t)round(abs((92+hitpos.X())*fPropDelay));
-        thit1 -= (uint64_t)round(abs((92+hitpos.X())*fPropDelay));
+    thit -= fSiPMtoFEBdelay;
+    thit1 -= fSiPMtoFEBdelay;
+    //thit1 = (Long64_t)(thit-GlobalTrigger[(int)mac+73]);
+    //thit -= (uint64_t)round(abs((92+hitpos.Z())*fPropDelay));
+    //std::cout<<"Hit position local: X "<<hitpos.X()<<" Z "<<hitpos.Z();
+    if((92-hitpos.X())<=(92+hitpos.Z())) {
+	//std::cout<<"  Thit raw: "<<thit<<" Correction (-hitposX) "<<thit - (uint64_t)round(abs((92+hitpos.X())*fPropDelay))<<"  Bar: "<<maxx*2<<"  "<<maxz*2<<std::endl;
+	thit -= (uint64_t)round(abs((92+hitpos.Z())*fPropDelay));
+	thit1 -= (uint64_t)round(abs((92+hitpos.Z())*fPropDelay));
+    }
+    else {
+	//std::cout<<"  Thit raw: "<<thit<<" Correction (-hitposZ) "<<thit - (uint64_t)round(abs((92+hitpos.Z())*fPropDelay))<<"  Bar: "<<maxx*2<<"  "<<maxz*2<<std::endl; 
+	thit -= (uint64_t)round(abs((92-hitpos.X())*fPropDelay));
+	thit1 -= (uint64_t)round(abs((92-hitpos.X())*fPropDelay));
+    }
```

Comment:
Is my reading right that this code translates into:
```suggestion
    double const corrPos = std::max(-hitpos.X(), hitpos.Z());
    uint64_t const corr = (uint64_t)round(abs((92.0+corrPos)*fPropDelay));
    thit -= corr;
    thit1 -= corr;
```

Reply:
You are reading right and your proposal looks more elegant (as always), thanks